# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: 62480de389ff
# Full Hash: 62480de389ffba353998de24760993b2e1531232
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth. CLOSED TREE
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r f2e486f1be17 -r 62480de389ff js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Thu Aug 27 10:30:46 2020 +0000
+++ b/js/src/wasm/WasmTypes.h	Thu Aug 27 10:53:04 2020 +0000
@@ -3258,7 +3258,6 @@
 
   uint8_t* rawCaller() const { return callerFP_; }
   TlsData* tls() const { return tls_; }
-  Instance* instance() const { return tls()->instance; }
 
   Frame* wasmCaller() const {
     MOZ_ASSERT(!callerIsExitOrJitEntryFP());
@@ -3414,7 +3413,7 @@
   static DebugFrame* from(Frame* fp);
   Frame& frame() { return frame_; }
   uint32_t funcIndex() const { return funcIndex_; }
-  Instance* instance() const { return frame_.instance(); }
+  Instance* instance() const;
   GlobalObject* global() const;
   bool hasGlobal(const GlobalObject* global) const;
   JSObject* environmentChain() const;
